src/game/Game.js

```diff
@@ -88,16 +78,53 @@ class Game {
     this.nameGraphic = new InstanceNameGraphic(this);
   }
 
+  startGamemode() {
+    // Create gamemode
+    const gamemodeHandler = GamemodeHandler.getInstance();
+    const { SelectedMode, resources, options } = gamemodeHandler.getSelected();
+    return new Promise(resolve => {
+      this.resourceServer.requestResources(resources).then(loadedResources => {
+        this.currentGamemode = new SelectedMode(this, loadedResources);
+        this.currentGamemode.init();
+
+        this.handler = new GamemodeConfigHandler(this, this.currentGamemode, options);
+
+        this.gamemodeLoaded = true;
+        this.notifyResizeListeners();
+        resolve();
+      });
+    });
+  }
+
+  switchGamemode(nextGamemode) {
+    const gamemodeHandler = GamemodeHandler.getInstance();
+    gamemodeHandler.selectGameMode(nextGamemode);
+    this.instance.stashPlayers();
+    this.gamemodeLoaded = false;
+
```

Comment:
If a player joins (and probably leaves) in the middle of a gamemode switch the player will not be recognized as a player (but is visible on the screen). When an already connected player collides with him the game will crash. Test this functionality by adding a player inside this function.

Also there is nothing that tells the controller/service that the gamemode has changes, so that functionality needs to be changed.+

Reply:
@Thraix I have fixed players joining during a switch. I have removed the ability to change the gamemode, now you can only restart the current one. (You are still able to switch manually through the browser console.)